src/structs/color.rs

```diff
@@ -1,111 +1,518 @@
+        a: 0xFF,
+        r: 0xFF,
+        g: 0xFF,
+        b: 0xFF,
+    };
+    pub const COLOR_WHITE_STR: &str = "FFFFFFFF";
+    pub const COLOR_YELLOW: ARGB8 = ARGB8 {
+        a: 0xFF,
+        r: 0xFF,
+        g: 0xFF,
+        b: 0x00,
+    };
+    pub const COLOR_YELLOW_STR: &str = "FFFFFF00";
+    pub const NAMED_COLORS: &[&str] = &[
         "Black", "White", "Red", "Green", "Blue", "Yellow", "Magenta", "Cyan",
     ];
 
-    // Colors
-    pub const COLOR_BLACK: &'static str = "FF000000";
-    pub const COLOR_WHITE: &'static str = "FFFFFFFF";
-    pub const COLOR_RED: &'static str = "FFFF0000";
-    pub const COLOR_DARKRED: &'static str = "FF800000";
-    pub const COLOR_BLUE: &'static str = "FF0000FF";
-    pub const COLOR_DARKBLUE: &'static str = "FF000080";
-    pub const COLOR_GREEN: &'static str = "FF00FF00";
-    pub const COLOR_DARKGREEN: &'static str = "FF008000";
-    pub const COLOR_YELLOW: &'static str = "FFFFFF00";
-    pub const COLOR_DARKYELLOW: &'static str = "FF808000";
+    /// Convert hex string to ARGB8
+    #[must_use]
+    pub fn hex_to_argb8(hex: &str) -> Option<ARGB8> {
+        if hex.len() == 9 && hex.starts_with('#') {
+            let a = u8::from_str_radix(&hex[1..3], 16).ok()?;
+            let r = u8::from_str_radix(&hex[3..5], 16).ok()?;
+            let g = u8::from_str_radix(&hex[5..7], 16).ok()?;
+            let b = u8::from_str_radix(&hex[7..9], 16).ok()?;
 
-    /// Get Argb.
-    /// If the color is based on the theme, it cannot be obtained with this function.
-    /// In that case, use get_argb_with_theme(&self, theme: &Theme).
-    pub fn get_argb(&self) -> &str {
-        if self.indexed.has_value() {
-            if let Some(v) = INDEXED_COLORS.get(self.indexed.get_value() as usize) {
-                return v;
+            Some(ARGB8 { a, r, g, b })
+        } else if hex.len() == 8 {
+            let a = u8::from_str_radix(&hex[0..2], 16).ok()?;
+            let r = u8::from_str_radix(&hex[2..4], 16).ok()?;
+            let g = u8::from_str_radix(&hex[4..6], 16).ok()?;
+            let b = u8::from_str_radix(&hex[6..8], 16).ok()?;
+
+            Some(ARGB8 { a, r, g, b })
+        } else if hex.len() == 7 && hex.starts_with('#') {
+            let a = 0xFF;
+            let r = u8::from_str_radix(&hex[1..3], 16).ok()?;
+            let g = u8::from_str_radix(&hex[3..5], 16).ok()?;
+            let b = u8::from_str_radix(&hex[5..7], 16).ok()?;
+
+            Some(ARGB8 { a, r, g, b })
+        } else if hex.len() == 6 {
+            let a = 0xFF;
+            let r = u8::from_str_radix(&hex[0..2], 16).ok()?;
+            let g = u8::from_str_radix(&hex[2..4], 16).ok()?;
+            let b = u8::from_str_radix(&hex[4..6], 16).ok()?;
+
+            Some(ARGB8 { a, r, g, b })
+        } else if hex.len() < 6 {
+            // To pass an integration test where the hex string is "#333".
+            // https://github.com/MathNya/umya-spreadsheet/pull/113
```

Comment:
I am not sure if this should be valid behaviour? Is `#333` really a valid color? I added padding so the previous integration test added in PR #113 still passes.

Reply:
This is the code.

Reply:
@mxsrm 
Hmmm, this seems to be a mistake for #333333.